projects/sandbox/sampling/sample.py

```diff
@@ -0,0 +1,167 @@
+import logging
+from pathlib import Path
+from time import time
+from typing import List
+
+import bilby
+import h5py
+import numpy as np
+import pandas as pd
+import torch
+
+from ml4gw.transforms import ChannelWiseScaler
+from mlpe.architectures import architecturize
+from mlpe.data.transforms import Preprocessor
+from mlpe.injection.priors import sg_uniform
+from mlpe.logging import configure_logging
+
+
+def _load_preprocessor_state(preprocessor, preprocessor_dir):
+    preprocessor_path = Path(preprocessor_dir) / "preprocessor"
+
+    whitener_path = preprocessor_path / "whitener.pt"
+    scaler_path = preprocessor_path / "scaler.pt"
+
+    preprocessor.whitener = torch.load(whitener_path)
+    preprocessor.scaler = torch.load(scaler_path)
+    return preprocessor
+
+
+def _load_test_data(testing_path: Path, inference_params: List[str]):
+    with h5py.File(testing_path, "r") as f:
+        signals = f["injections"][:]
+        params = []
+        for param in inference_params:
+            values = f["phase" if param == "phi" else param][:]
+            # take logarithm since hrss
+            # spans large magnitude range
+            if param == "hrss":
+                values = np.log10(values)
+            params.append(values)
+
+        params = np.vstack(params).T
+    return signals, params
+
+
+def _cast_as_bilby_result(samples, truth, inference_params, priors):
+    """Cast samples as bilby Result object"""
+    # samples shape (1, num_samples, num_params)
+    # inference_params shape (1, num_params)
+    samples = samples[0]
+    truth = truth[0]
+    injections = {k: float(v) for k, v in zip(inference_params, truth)}
+
+    posterior = dict()
+    for idx, k in enumerate(inference_params):
+        posterior[k] = samples.T[idx].flatten()
+    posterior = pd.DataFrame(posterior)
+    return bilby.result.Result(
+        label="test_data",
+        injection_parameters=injections,
+        posterior=posterior,
+        search_parameter_keys=inference_params,
+        priors=priors,
+    )
+
+
+@architecturize
+def main(
+    architecture: callable,
+    model_state_path: Path,
+    ifos: List[str],
+    sample_rate: float,
+    trigger_distance: float,
+    kernel_length: float,
+    fduration: float,
+    inference_params: List[str],
+    datadir: Path,
+    logdir: Path,
+    outdir: Path,
+    device: str,
+    num_samples_draw: int,
+    verbose: bool = False,
+):
+    device = device or "cpu"
+    # FIXME: fix phi/phase discrepancy
+    priors = sg_uniform()
+    priors["phi"] = priors["phase"]
+    del priors["phase"]
+
+    configure_logging(logdir / "sample.log", verbose)
+    num_ifos = len(ifos)
+    num_params = len(inference_params)
+
+    logging.info("Initializing model and setting weights from trained state")
+    model_state = torch.load(model_state_path)
+    flow_obj = architecture((num_params, num_ifos, 2048))
```

Comment:
Can we "un-hard code" this: I think it should be `(kernel_length - f_duration) * sample_rate`